flink-1.17-connector-bigquery/flink-connector-bigquery/src/test/java/com/google/cloud/flink/bigquery/sink/serializer/RowDataToProtoSerializerTest.java

```diff
@@ -0,0 +1,517 @@
+package com.google.cloud.flink.bigquery.sink.serializer;
+
+import org.apache.flink.table.data.GenericArrayData;
+import org.apache.flink.table.data.GenericRowData;
+import org.apache.flink.table.data.StringData;
+import org.apache.flink.table.data.TimestampData;
+import org.apache.flink.table.types.logical.LogicalType;
+
+import com.google.cloud.flink.bigquery.sink.exceptions.BigQuerySerializationException;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.Descriptors;
+import com.google.protobuf.DynamicMessage;
+import org.apache.avro.Schema;
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static com.google.cloud.flink.bigquery.sink.serializer.TestBigQuerySchemas.getAvroSchemaFromFieldString;
+import static com.google.cloud.flink.bigquery.sink.serializer.TestBigQuerySchemas.getRecordSchema;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+
+/** Test to check the RowData to proto Serializer. */
+public class RowDataToProtoSerializerTest {
+
+    /**
+     * Test to check <code>getDynamicMessageFromRowData()</code> for Primitive types supported by
+     * BigQuery.
+     *
+     * <ul>
+     *   <li>BQ Type - Converted Table API Schema Type
+     *   <li>"INTEGER" - LONG
+     *   <li>"FLOAT" - FLOAT
+     *   <li>"STRING" - STRING
+     *   <li>"BOOLEAN" - BOOLEAN
+     *   <li>"BYTES" - BYTES
+     * </ul>
+     */
+    @Test
+    public void testAllBigQuerySupportedPrimitiveTypesConversionToDynamicMessageCorrectly()
```

Comment:
does this test actually test all the supported primitive types? what about date, timestamps, etc? maybe i'm misunderstanding the test

Reply:
if i'm understanding correctly, looks like maybe the test on line 374 covers this

Reply:
By primitive types i meant types such as `INTEGER`, `STRING`, `BOOLEAN`, `BYTES`, `FLOAT` etc.

Reply:
Line 374 covers the leftover types (Logical Types in avro)  such as `NUMERIC`, `BIGNUMERIC`, `DATETIME`,  `DATE`, `TIME` and `TIMESTAMP`

Reply:
ah got it. you can resolve this convo